internal/server/server.go

```diff
@@ -20,14 +24,38 @@ var (
 
 // Server contains all the necessary information to run Bifrost
 type Server struct {
-	cfg Config
-	srv *http.Server
+	cfg       Config
+	srv       *http.Server
+	logger    *log.Logger
+	client    *http.Client
+	getHostFn func(bucket, endPoint string) string
+	creds     *credentials.Credentials
 }
 
 // New creates a new Bifrost server
 func New(cfg Config) *Server {
+	// All settings are same as DefaultTransport,
+	// with MaxConnsPerHost and ResponseHeaderTimeout added.
+	client := &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxConnsPerHost:       cfg.ServiceSettings.MaxConnsPerHost,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ResponseHeaderTimeout: 30 * time.Second,
```

Comment:
```suggestion
			ResponseHeaderTimeout: time.Duration(cfg.ServiceSettings.ResponseHeaderTimeout) * time.Second,
```

Reply:
Yeah, I didn't really see a need for response header timeouts to be configurable. Hence didn't keep it. But we can include it if we need it later.